contrib/babelfishpg_common/src/datetime.c

```diff
@@ -196,7 +196,7 @@ clean_input_str(char *str, bool *contains_extra_spaces, DateTimeContext context)
 		if (str[i] == '\0')
 			break;
 
-		if (j > MAXDATELEN)
+		if (j >= MAXDATELEN)
```

Comment:
We are also doing j+=2 below.
Is this condition enough to make sure we do not overflow there ?

Reply:
We are also adding a null byte at the end after the while loop ends. So should this be be MAXDATALEN - 1.
i.e. not writing on the last byte inside the while loop.

Reply:
> We are also doing j+=2 below. Is this condition enough to make sure we do not overflow there ?

Yes, this condition should be sufficient to make sure that there is no overflow.

> We are also adding a null byte at the end after the while loop ends. So should this be be MAXDATALEN - 1.
i.e. not writing on the last byte inside the while loop.

We can update the size of result to MAXDATALEN + 1 instead.